eva_submission/vep_utils.py

```diff
@@ -0,0 +1,222 @@
+import ftplib
+import os
+import re
+import tarfile
+from fnmatch import fnmatch
+
+import pymongo
+import requests
+from retry import retry
+
+from ebi_eva_common_pyutils.config import cfg
+from ebi_eva_common_pyutils.logger import logging_config as log_cfg
+
+
+logger = log_cfg.get_logger(__name__)
+
+ensembl_ftp_url = 'ftp.ensembl.org'
+ensembl_genome_ftp_url = 'ftp.ebi.ac.uk'
+ensembl_genome_dirs = [
+    'ensemblgenomes/pub/plants',
+    'ensemblgenomes/pub/metazoa',
+    'ensemblgenomes/pub/fungi',
+    'ensemblgenomes/pub/protists',
+    'ensemblgenomes/pub/bacteria',
+]
+
+# Name of collection in variant warehouse to check for existing VEP versions
+annotation_collection_name = 'annotations_2_0'
+
+
+def vep_path(version):
+    return os.path.join(cfg['vep_path'], f'ensembl-vep-release-{version}/vep')
+
+
+def get_vep_and_vep_cache_version(mongo_uri, db_name, taxonomy_id, assembly_accession):
+    """
+    Gets VEP and VEP cache versions for a given assembly by first checking what is already in the variant DB,
+    then checking Ensembl and Ensembl Genome FTPs, otherwise returns None.
+    """
+    vep_version, vep_cache_version = get_vep_and_vep_cache_version_from_db(mongo_uri, db_name)
+    if not vep_cache_version and not vep_version:
+        vep_version, vep_cache_version = get_vep_and_vep_cache_version_from_ensembl(
+            db_name, taxonomy_id, assembly_accession)
+
+    if check_vep_version_installed(vep_version):
+        return vep_version, vep_cache_version
+    raise ValueError(
+        f'Found VEP cache version {vep_cache_version} for assembly {assembly_accession}, '
+        f'but compatible VEP version {vep_version} is not installed.'
+    )
```

Comment:
If nothing is found in Ensembl the ValueError is raised here. 
Either let  `check_vep_version_installed` return True when `vep_version is None` or you need to first return `(None, None)` if the  `vep_version` or `vep_cache_version` are not found.